### DISCUSSION TITLE
Using IPN vs Name for Parts - Why both?

### DISCUSSION THREAD

**Discussion by motusRP** [inventree/InvenTree#7068] [Q&A]
I'm curious how people are using IPN and Name in Inventree.  We are a small-er company that does consulting work and we're looking for a way to manage parts, inventory, builds, etc. and Inventree is looking like *exactly* what we need.

Looking through the Demo site, it seems that "Name" has been used either as generic words (Blue Table, Leg, Top, Chair, etc.) or as an abbreviated description (R_220k_0402_1%) that is expanded in the Description field into real english.

We do have an existing part numbering (IPN) system (it's a first cut, open for revision) that has a aa-bbbbb-ccc format where all digits are numbers.
- aa = category (mechanical, electrical, marketing, document, etc.)
- bbbbb = project (just a sequential 5 digit number)
- ccc = sequential number for the part

Descriptions on existing parts are a comma-separated list of keywords.  ie.
- Assembly, Table, 4-person, Blue
- Leg, Table, Turned, Red, 3ft
- Screw, SHCS, M4x20mm, 3mm Hex, SS316, Black Oxide

It looks like we would be better off putting our IPN into the Name field, and leaving IPN blank, because it's unlikely that we'd be able to come up with unique Names for everything we had that weren't just coded text in some way.  At least an IPN system lets us create those names in an organized way and navigate them easily later.  I can see how "Name" might work for a really small company with a few assemblies to manage, but if you expand to a company with 10's or 100's of assemblies and 100's or 1000's of parts, generating a unique name for each one will become impossible *without* an IPN system (or just using sequential numbers for parts... ugh).

I did do an import with Name and IPN matching, but that becomes awkward as Inventree seems to combine Name, IPN, and Description in lists, so you end up with things like this:
- 01-00001-001 | 01-00001-001 | Assembly, Table, 4-Person, Blue
Where the number is duplicated unnecessarily.  I tried leaving "Name" blank on import but it didn't like that.

Am I treading ground that others have treaded before?  Or am I missing something fundamental?

Thanks in advance!

**Comment by LastEdit** [comment]
I went through the exact same thing. I had an IPN and a description. The Name in the way it is presented could be anything, but like you said, it feels very much like a description in the demo. I don't think there is a best way to implement it. I imported IPNs into both fields since the Name field is required and left the description as is.
The only thing I do is customize the way it is presented. If you are comfortable with minor coding you can edit all the reports to only display either the Name or IPN instead of the fullname. 
You can also change the way it is presented in the site under the Part Name column. They have a customizable setting under Settings -> Parts -> Part Name Display Format which you can tweak so only the Part Name or IPN will show depending on what you want. Obviously if you have IPNs in both fields like I do right now, you should remove one of the columns by toggling it off.
The reason I brought it in both columns is if I ever figure out a different use for the Name field, I have the IPN already imported. But I doubt I will ever use the Name field as intended.

**Comment by matmair** [comment]
@motusRP Name and IPN have different restrictions and purposes.
Name has only a max length and is meant to be shown to the user. 
IPN is normally enforced to be unique (that can be controlled via PART_ALLOW_DUPLICATE_IPN) and meant to be used as an identifier.

As @LastEdit pointed out, you can control the format part names are displayed in.

**Reply by motusRP** [reply]
It appears to me that Name must be unique as well between parts... Do I have that wrong?

**Reply by motusRP** [reply]
Correction, it seems Name does not need to be unique.  Is there a way to make Inventree require unique Names as well as IPN's?

**Reply by matmair** [reply]
No, that is what IPNs are designed for

**Comment by SergeoLacruz** [comment]
We also ise IPN in a fixed format like xxx.yyyyy.zzzz where each number is unique. The name in our case is the manufacturer part number. We have a 1:1 correlation between them. E.g. 1k resistor from KOA has adifferent IPN comared to a 1k resistor from Vishay.

**Reply by motusRP** [reply]
You're putting the Manufacturer's Part Number (MPN) in the Name field, but are you also putting the MPN in the Manufacturer's Part list and tying it to the IPN that way?

**Reply by SergeoLacruz** [reply]
Yes. Sometimes we have serveral MPN in the supplier part list. E.g. different codes for different reel sizes or other pack options loike
tray vs. reel.
